docs/Routes.md

```diff
@@ -51,8 +51,8 @@ They need to be in
 * `onSend(request, reply, payload, done)`: a [function](https://github.com/fastify/fastify/blob/master/docs/Hooks.md#route-hooks) called right before a response is sent, it could also be an array of functions.
 * `onResponse(request, reply, done)`: a [function](https://github.com/fastify/fastify/blob/master/docs/Hooks.md#onresponse) called when a response has been sent, so you will not be able to send more data to the client. It could also be an array of functions.
 * `handler(request, reply)`: the function that will handle this request.
-* `validatorCompiler(method, url, httpPart, schema)`: function that builds schemas for request validations. See the [Validation and Serialization](https://github.com/fastify/fastify/blob/master/docs/Validation-and-Serialization.md#schema-validator) documentation.
-* `serializerCompiler(method, url, httpPart, schema)`: function that builds schemas for response serialization. See the [Validation and Serialization](https://github.com/fastify/fastify/blob/master/docs/Validation-and-Serialization.md#schema-serializer) documentation.
+* `validatorCompiler(schema, method, url, httpPart)`: function that builds schemas for request validations. See the [Validation and Serialization](https://github.com/fastify/fastify/blob/master/docs/Validation-and-Serialization.md#schema-validator) documentation.
```

Comment:
```suggestion
* `validatorCompiler({schema, method, url, httpPart})`: function that builds schemas for request validations. See the [Validation and Serialization](https://github.com/fastify/fastify/blob/master/docs/Validation-and-Serialization.md#schema-validator) documentation.
```

If you're going to change it, might as well make it easier to extend.

Reply:
Not sure it will need to be more extensible, I can't imagine a new parameter that would be needed right now, still thinking

Reply:
1. Changing the signature just to change the position of a parameter is going to cause a lot of confusion (and probably no shortage of GitHub issues)
2. I'd rather see a destructured object as the parameter because it is easier to read, doesn't require a lot of checking and re-assigning of parameters, and will be an easier change over all